bpfman/src/lib.rs

```diff
@@ -1801,25 +1801,29 @@ fn detach_multi_attach_program(
     } else {
         false
     };
-    debug!("netns_deleted = {netns_deleted}");
+    let number_attached_programs = num_attached_programs(&did, root_db)?;
+    debug!(
+        "netns_deleted = {netns_deleted}, number_attached_programs = {number_attached_programs}"
+    );
 
-    let next_available_id = if netns_deleted {
+    // Typically, we'd expect the number of attached programs to be greater than
+    // 0 because we're calling the detach function.  However, if
+    // num_attached_programs() can't find the dispatcher for the given did, it
+    // will return 0, so we need to handle that case.
+    let next_available_id = if number_attached_programs == 0 {
         0
     } else {
-        num_attached_programs(&did, root_db)? - 1
+        number_attached_programs - 1
     };
     debug!("next_available_id = {next_available_id}");
 
     let mut old_dispatcher = get_dispatcher(&did, root_db)?;
 
-    if let Some(ref mut old) = old_dispatcher {
-        if next_available_id == 0 {
+    if next_available_id == 0 || netns_deleted {
```

Comment:
Do we even need `netns_deleted` at all?
If so, a comment here explaining this condition would be helpful.
Perhaps move the comment from above to here where it's more relevant.

Reply:
Done.